src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs

```diff
@@ -121,18 +133,24 @@ public void SetExitCallback(int processId, Action callbackAction)
 
             process.EnableRaisingEvents = true;
             process.Exited += (sender, args) =>
-                {
-                    callbackAction.Invoke();
-                };
+            {
+                callbackAction.Invoke();
+            };
         }
 
         /// <inheritdoc/>
         public void TerminateProcess(object process)
         {
             var proc = process as Process;
-            if (proc != null && !proc.HasExited)
+            try
+            {
+                if (proc != null && !proc.HasExited)
+                {
+                    proc.Kill();
```

Comment:
This can return `Win32Exception` as well if the process is terminating?
Shouldn't we catch that too?